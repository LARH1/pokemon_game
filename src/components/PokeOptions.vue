<template>
<h4>Selecciona una opci√≥n</h4>
<div class="options-container">
    <ul>
        <li :key=poke.id v-for="poke in options"
        @click="$emit('poke-selection',poke.id)">{{poke.name}}</li>
    </ul>
</div>
</template>

<script>
export default
{
    emits: ["poke-selection"], // Definir emits
    props:
    {
        options:
        {
            type: Array,
            required: true,
        }
    }
}
</script>

<style scoped>
.options-container {
    display: flex;
    justify-content: center;
}

ul {
    text-align: center;
    list-style-type: none;
}

li {
    border: 1px solid rgba(203, 201, 201, 0.5);
    cursor: pointer;
    margin-bottom: 10px;
    padding: 10px 5px 5px 5px;
    border-radius: 8px;
    width: 250px;
    background-color: white;
}

li:hover {
    background-color: aliceblue;
}
</style>
