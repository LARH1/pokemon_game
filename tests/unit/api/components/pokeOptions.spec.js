import PokeOptions from "@/components/PokeOptions";
import { shallowMount } from "@vue/test-utils";

describe("pokeOptions",() =>
{
    let wrapper;
    // Definir el snapshot para usarlo global
    beforeEach(() =>
    {
        wrapper = shallowMount(PokeOptions,
            {
                props:
                {
                    options: [
                        {id: 1,name: "BULBASAUR"},
                        {id: 2,name: "IVYSAUR"},
                        {id: 3,name: "VENUSAUR"},
                        {id: 4,name: "CHARMANDER"},
                    ]
                }
            });
    });

    test("Debe coincidir con el snapshot",() =>
    {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("Debe contener 4 elementos y coincidir con el nombre",() =>
    {
        const data = wrapper.findAll("li");
        expect(data.length).toBe(4); // 4 Elementos
        // .text() -> obtiene el texto del elemento
        expect(data[0].text()).toBe("BULBASAUR");
        expect(data[1].text()).toBe("IVYSAUR");
        expect(data[2].text()).toBe("VENUSAUR");
        expect(data[3].text()).toBe("CHARMANDER");
    });

    test("Debe emitir el evento 'poke-selection' con sus parametros",() =>
    {
        const [l1,l2,l3,l4] = wrapper.findAll("li");
        // Mandar evento
        l1.trigger("click");
        l2.trigger("click");
        l3.trigger("click");
        l4.trigger("click");
        // Argumento esperado:1,2,3,4
        // console.log(wrapper.emitted("poke-selection"));
        expect(wrapper.emitted("poke-selection")[0][0]).toBe(1);
        expect(wrapper.emitted("poke-selection")[1][0]).toBe(2);
        expect(wrapper.emitted("poke-selection")[2][0]).toBe(3);
        expect(wrapper.emitted("poke-selection")[3][0]).toBe(4);
    });
})