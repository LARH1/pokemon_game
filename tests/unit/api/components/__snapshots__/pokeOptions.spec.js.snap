// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokeOptions Debe coincidir con el snapshot 1`] = `
<h5 class="mt-2">Selecciona una opci√≥n</h5>
<div class="options-container">
  <ul class="asd">
    <li>BULBASAUR</li>
    <li>IVYSAUR</li>
    <li>VENUSAUR</li>
    <li>CHARMANDER</li>
  </ul>
</div>
`;
